### PR TITLE
forgejo-actions-runner: 3.0.1 -> 3.1.0

### PR DESCRIPTION
**Pull request by yayayayaka**
## Description of changes
https://code.forgejo.org/forgejo/runner/src/tag/v3.1.0/RELEASE-NOTES.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by abueide** [comment]
is anyone able to run an action within a container? For example with this action:

```
name: Deploy Systems

on:
  push:
    branches:
      - main

jobs:
  deploy:
    runs-on: self-hosted
    container: 
      image: nixos/nix:latest

    steps:
    - name: Setup SSH Key
      run: |
        mkdir -p ~/.ssh
        echo "${{ secrets.DEPLOYRS_KEY }}" > ~/.ssh/id_rsa
        chmod 600 ~/.ssh/id_rsa
        ssh-keyscan -H [hostname] >> ~/.ssh/known_hosts
      env:
        DEPLOYRS_KEY: ${{ secrets.DEPLOYRS_KEY }}

    - name: Checkout Repository
      uses: https://github.com/actions/checkout@v3

    - name: Install deploy-rs
      run: |
        nix-shell -p deploy-rs

    - name: Deploy Infrastructure
      run: |
        deploy infra/nix/systems/manjiro
```

i'm getting this error on the "setup job" phase

```
 🐳  docker run image=nixos/nix:latest platform= entrypoint=["/bin/sleep" "10800"] cmd=[]
Starting container: 8606995fe7353a098bb6eec2a5295b0254413887b4b55d367f612cc133edd4bd
failed to start container: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/bin/sleep": stat /bin/sleep: no such file or directory: unknown
```
Is there a way to link it to the binary its expecting? will this require an upstream patch?

**Comment by Janik-Haag** [comment]
@abueide is this a regression or was it already not working with older versions?

**Comment by Janik-Haag** [comment]
The container probably fails to start since there is no /bin/sleep only:
image

**Comment by abueide** [comment]
> The container probably fails to start since there is no /bin/sleep only: image

I'm pretty sure it exists in old versions, not a regression not too sure though.

**Comment by Janik-Haag** [comment]
I would say this is fine to merge then. wdyt @yayayayaka ?

File: pkgs/development/tools/continuous-integration/forgejo-actions-runner/default.nix [hunk 1]
```diff
@@ -7,17 +7,17 @@
 
 buildGoModule rec {
   pname = "forgejo-actions-runner";
-  version = "3.0.1";
+  version = "3.1.0";
```
Comment (teutat3s) [inline pkgs/development/tools/continuous-integration/forgejo-actions-runner/default.nix:10]: Version [3.3.0](https://codeberg.org/forgejo/runner/src/branch/main/RELEASE-NOTES.md#330) is out by now, are you able to bump to that version here?

File: pkgs/development/tools/continuous-integration/forgejo-actions-runner/default.nix [hunk 2]
```diff
@@ -7,17 +7,17 @@
...
 
   src = fetchFromGitea {
     domain = "codeberg.org";
```
Comment (teutat3s) [inline pkgs/development/tools/continuous-integration/forgejo-actions-runner/default.nix:13]: The upstream repo is now https://code.forgejo.org/forgejo/runner (keeping the read-only mirror would be fine for me though).

**Review by teutat3s — COMMENTED** [review]
This looks good to go for me, just two nits.

**Comment by yayayayaka** [comment]
Sorry, I was on holidays, but thanks for taking care of it :heart:

**Pull request closed without merging**